src/server/unix.rs

```diff
@@ -482,9 +482,16 @@ fn _reinit_and_restore(
         .join(format!("{}.dump", inst.name()));
 
     cmd.background_for(async {
+        let builder = inst.get_connector(true)?;
         upgrade::restore_instance(
-            inst, &dump_path, inst.get_connector(true)?
+            inst, &dump_path, builder.clone()
         ).await?;
+
+        // TODO: skip if --non-interactive
+        credentials::ask_update_credentials(
```

Comment:
I think this also doesn't require asking user, as described earlier. Also this doesn't need to be done in `background_for` block.